contracts/examples/erc1155-marketplace/src/lib.rs

```diff
@@ -0,0 +1,386 @@
+		self.set_percentage_cut(bid_cut_percentage);
+	}
+
+	// endpoints - Token ownership contract only
+
+	/// `_operator` argument is ignored, but it has to be kept because of the erc1155 standard
+	#[endpoint(onERC1155Received)]
+	fn on_erc1155_received(
+		&self,
+		_operator: Address,
+		from: Address,
+		type_id: BigUint,
+		token_id: BigUint,
+		args: AuctionArgument<BigUint>,
+	) -> SCResult<u32> {
+		require!(
+			self.get_caller() == self.get_token_ownership_contract_address(),
+			"Only the token ownership contract may call this function"
+		);
+
+		sc_try!(self.try_create_auction(
+			&type_id,
+			&token_id,
+			&from,
+			&args.min_bid,
+			&args.max_bid,
+			args.deadline
+		));
+
+		Ok(ACCEPTED_TRANSFER_ANSWER)
+	}
+
+	/// Same `AuctionArgument` is used for all tokens  
+	/// `_operator` argument is ignored, but it has to be kept because of the erc1155 standard
+	#[endpoint(onERC1155BatchReceived)]
+	fn on_erc1155_batch_received(
+		&self,
+		_operator: Address,
+		from: Address,
+		type_ids: Vec<BigUint>,
+		token_ids: Vec<BigUint>,
+		args: AuctionArgument<BigUint>,
+	) -> SCResult<u32> {
+		require!(
+			self.get_caller() == self.get_token_ownership_contract_address(),
+			"Only the token ownership contract may call this function"
+		);
+		require!(
+			type_ids.len() == token_ids.len(),
+			"type_ids and token_ids lengths do not match"
+		);
+
+		// Don't have to worry about checking if there are duplicates in the entries,
+		// an error here will revert all storage changes automatically
+		for (type_id, token_id) in type_ids.iter().zip(token_ids.iter()) {
+			sc_try!(self.try_create_auction(
+				type_id,
+				token_id,
+				&from,
+				&args.min_bid,
+				&args.max_bid,
+				args.deadline
+			));
+		}
+
+		Ok(ACCEPTED_TRANSFER_ANSWER)
+	}
+
+	// endpoints - owner-only
+
+	#[endpoint]
+	fn claim(&self) -> SCResult<()> {
+		only_owner!(self, "Only owner may call this function!");
+
+		let claimable_funds = self.get_claimable_funds();
+		self.set_claimable_funds(&BigUint::zero());
+		
+		self.send()
+			.direct_egld(&self.get_caller(), &claimable_funds, b"claim");
+
+		Ok(())
+	}
+
+	#[endpoint(setCutPercentage)]
+	fn set_percentage_cut_endpoint(&self, new_cut_percentage: u8) -> SCResult<()> {
+		only_owner!(self, "Only owner may call this function!");
+		require!(
+			new_cut_percentage > 0 && new_cut_percentage < PERCENTAGE_TOTAL,
+			"Invalid percentage value, should be between 0 and 100"
+		);
+
+		self.set_percentage_cut(new_cut_percentage);
+
+		Ok(())
+	}
+
+	fn set_token_ownership_contract_address_endpoint(&self, new_address: Address) -> SCResult<()> {
+		only_owner!(self, "Only owner may call this function!");
+		require!(!new_address.is_zero(), "Cannot set to zero address");
+		// TODO: Also check the address with IsSmartContractAddress() once it's added to the API
+
+		self.set_token_ownership_contract_address(&new_address);
+
+		Ok(())
+	}
+
+	// endpoints
+
+	#[payable("EGLD")]
```

Comment:
it should receive ESDT as well, and verify if auction accepts that kind of token.

Reply:
Done